adoc/admin-cluster-management.adoc

```diff
@@ -0,0 +1,125 @@
+
+- The `--control-plane` flag specifies the IP/FQDN of your load balancer.
+If you do not use a load balancer use the IP/FQDN of your first master node.
+- Substitute `<cluster-name>` for a suitable name for your cluster and hence the resulting directory.
+
+
+=== Bootstrapping the cluster
+
+To add the first master node use the `skuba node bootstrap` command:
+
+[source,bash]
+skuba node bootstrap --user <user-name> --sudo --target <IP/FQDN> <master-node>
+
+The mandatory flags for this command are `--user`, `--sudo` and `--target`.
+Provide the following information:
+- `--sudo` is for running the command with superuser privileges,
+this is required for all node operations.
+- `<user-name>` is the instance user name defined in the `terraform.tfvars` file.
+- `<IP/FQDN>` is the IP address or FQDN of the relevant virtual machine.
+- Substitute `<master-node>` for a unique master node name.
+
+
+== Adding nodes
+
+Once you have added the first master node to the cluster,
+use the `skuba node join` command to add more nodes.
+
+[source,bash]
+skuba node join --role <master/worker> --user <user-name> --sudo --target <IP/FQDN> <node-name>
+
+The mandatory flags for the join command are `--role`, ``--user`, `--sudo` and `--target`.
+
+- `--role` serves to specify if the node is a *master* or *worker*
+- `--sudo` is for running the command with superuser privileges.
+- `<user-name>` is the instance user name defined in the `terraform.tfvars` file.
+- `<IP/FQDN>` is the IP address or FQDN of the relevant virtual machine.
+- Substitute `<node-name>` for a unique node name.
+
+For example to add a *new master node* you would run something like this:
+
+[source,bash]
+skuba node join --role master --user sles --sudo --target 10.86.2.165 master2
+
+To add a *new worker node*, the process is essentially the same:
+
+[source,bash]
+skuba node join --role worker --user sles --sudo --target 10.86.2.164 worker1
+
+[TIP]
+====
+Use `skuba cluster status` to check that nodes have been successfully joined.
+====
+
+
+== Removing nodes
+
+The `skuba node remove` command serves to *permanently* remove nodes.
+Running this command will work even if the target virtual machine is down,
+so it is the safest way to remove the node.
+
+[source,bash]
+skuba node remove <node-name>
+
+If you wish to remove the node and want to add it again, the recommended approach is to
+first drain the node using `kubectl`. This will also cordon the node,
+so that it doesn't receive any new workloads.
+When you want to bring the node back, you only have to uncordon it.
+Refer to the official Kubernetes documentation for more information:
+https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/#use-kubectl-drain-to-remove-a-node-from-service
+
+
+== Resetting nodes
+
+The `skuba node reset` command enables you to reset a node to a state prior to join or bootstrap being run.
+Use this command to retry a failed bootstrap or join on a node or if the node fails.
```

Comment:
i'd say "or if the node has to be evacuated from the cluster"

Reply:
To me this is not clear. If I am going to remove a node from the cluster why would I have to reset it as well?

Reply:
@hwoarang, well that makes two of us. :) @ereslibre, could you please review this?

Reply:
Resetting a node should only be performed if either `bootstrap` or `join` failed on that node when trying to bootstrap a cluster or join this machine to an existing cluster. Reset should not be executed in any other context.

Since `bootstrap` and `join` are **not** idempotent actions, mainly because `kubeadm init` and `kubeadm join` (used internally) are not idempotent, when a machine fails to join in an unspecified state, we need to reset it before attempting the action again.